### PR TITLE
Removed href from Tabs links.

### PR DESCRIPTION
**Pull request by wachuga**

### CODE REVIEW

**Comment by wachuga** [comment]
This PR corrects the imageto appear as should under high contrast

Please see
https://identitydivision.visualstudio.com/OData/_workitems/edit/1366629

Note: User credentials should NOT be included in the bug.
Repro Steps:
Hit the URL https://www.odata.org/ and login with appropriate credentials to open Odata website
Tab till 'Developers' link and press enter
Tab till 'Reference Services' option and press enter
Verify all the controls in the "Odata V4" are accessible and Switch to high contrast theme and verify the issue.
Actual Result:
Tab button control is not adapting high contrast theme under Reference Services-Odata V4 blade.

Expected Result:
Tab button control should adapt high contrast theme under Reference Services-Odata V4 blade.

Note: Same issue exists throughout the application for all tab controls.

The tabs currently, (Normal mode)
After

Tabs after change (Normal mode)
image

tabs currently (High contrast)
before

Tabs after change (High contrast)
image

**Review by habbes — CHANGES_REQUESTED** [review]
@wachuga based on the screenshots. It seems your PR also changes how the tabs look in "normal" mode (i.e. when high contrast theme is not applied). Was this necessary?

Also the description states that this issues affects tabs all across the site. But your fix seems to apply to only the 3 tabs on the references page. I think it should be applied to all pages with tabs.

**Comment by habbes** [comment]
@wachuga I also have another question based on the original issue description and repro:
image

It says that tab buttons do not adapt to the high-contrast theme. But the theme settings in the above screenshots do state that hyperlinks should be yellow, which is what's currently the case. Should tab buttons not be considered as hyperlinks?

I'm comparing with how this GitHub page handles high-contrast settings. The main PR tab buttons have yellow text:
image

But the tab buttons when editing a comment have white text:
image

I'm not sure what the right approach is, I think tab buttons are links and ought to be treated as links. But if you decide to keep the as buttons, then I think you should apply it to all the tabs across the site.

**Comment by wachuga** [comment]
@habbes  I think the issue is that the tabs should not be links.

**Review by marabooy — APPROVED** [review]

**Review by paulodero — APPROVED** [review]

**Comment by habbes** [comment]
> @habbes I think the issue is that the tabs should not be links.

I'm confused by that statement, since it seems that you kept the tabs as links. Could you explain how your changes work?

**Comment by habbes** [comment]
And what about the tabs on other pages, like https://www.odata.org/getting-started/understand-odata-in-6-steps/

**Review by wandeg — COMMENTED** [review]
Has this change been tested to make sure it doesn't break existing accessibility fixes for the Narration?

**Comment by marabooy** [comment]
@wandeg Could you test out some of those features. I tested navigation and a bit of naration. Ideally what I have seen is that tabs.js takes care of tab activations as well and the current narration should not be affected by the changes. We should however see if we are firing any js events that bootstrap was firing before.

File: pages/ecosystem.html [hunk 1]
```diff
@@ -4,19 +4,21 @@
...
+
 <div class="row">
 <div class="col-md-2">
-<ul class="nav nav-pills nav-stacked">
-	<li class="active"><a href="#producers" data-toggle="pill">Producers</a></li>
-	<li><a href="#consumers" data-toggle="pill">Consumers</a></li>
-	<li><a href="#liveservices" data-toggle="pill">Live Services</a></li>
-	<li><a href="#sdk" data-toggle="pill">Sample Code</a></li>
-	<li><a href="#tutorials" data-toggle="pill">Tutorials</a></li>
+<ul class="nav nav-pills nav-stacked" target="ecosystem" >
```
Comment (marabooy) [inline pages/ecosystem.html:10]: Target here is used wrongly you can use custom tags which do not change the markup and are not reserved.
e.g. data-target-tab or something of the sort.

Same for the other items in the anchors as the value for target is specified [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target) and some browsers may by default have unspecified behaviours for unknown values here. a target=something implies there is an iframe named something.

**Review by marabooy — APPROVED** [review]

**Review by marabooy — CHANGES_REQUESTED** [review]
Kindly address the anchor targets and use custom attributes.

File: _includes/footer.html [hunk 2]
```diff
@@ -4,4 +4,20 @@
...
+   
+   var i, tabcontent, tablinks;
+   tabcontent = document.getElementsByClassName(panelclassName);
+   for (i = 0; i < tabcontent.length; i++) {
+    tabcontent[i].classList.remove("active");
+  }
+  document.getElementById(target).className  = "tab-pane active " + panelclassName;
+  
+});
+</script>
```
Comment (marabooy) [inline _includes/footer.html:21]: Move to _scripts.html

**Review by marabooy — COMMENTED** [review]
Move items to scripts include.

**Comment by habbes** [comment]
I've tested the changes with the narrator and keyboard navigation and found that it does not behave like the current site. 

When you tab through the pages using the Tab key, it skips the tab components on the page altogether. But when you navigate using arrows keys, then you can reach the tab items.

On the current site, you can reach the tab items using the Tab key.

I also noticed that there doesn't seem to be any UI indicator of the currently selected tab  item. But this seems to be an issue with the existing site as well. For example, looking at the image below, you can't easily tell which tab item is selected. But maybe that's beyond the scope of this PR.

image

**Review by marabooy — APPROVED** [review]

**Review by habbes — CHANGES_REQUESTED** [review]
I've tested with Narrator + Tabbing through page using the tab key and found the following issues:
- The sidebar links/tabs on the ecosystem page seems to be broken: clicking a link does not open the corresponding tab contents. Furthermore, when tabbing through the page, these links are skipped.
- On the Getting Started in 6 Easy Steps page, when you tab through, you eventually land on the first tab item, as expected. However when you continue to tab, it skips to the  "View Response" button without visiting the other tab items. From there it continues jumping to the next "View Response" button skipping the subsequent tab items in the process. In the original site, tabbing through the page visits all the tab items.
- On the References page, it also visits just visit the tab item when you tab through the page. But this seems to be happening in the original site as well.

**Comment by habbes** [comment]
> I've tested with Narrator + Tabbing through the page using the tab key and found the following issues:
> 
> * The sidebar links/tabs on the ecosystem page seems to be broken: clicking a link does not open the corresponding tab contents. Furthermore, when tabbing through the page, these links are skipped.
> * On the Understanding OData in 6 Easy Steps page, when you tab through, you eventually land on the first tab item, as expected. However when you continue to tab, it skips to the  "View Response" button without visiting the other tab items. From there it continues jumping to the next "View Response" button skipping the subsequent tab items in the process. In the original site, tabbing through the page visits all the tab items.
> * On the References page, it also visits just visit the tab item when you tab through the page. But this seems to be happening in the original site as well.

Seems that I had not updated the branch on my local tests. My mistake. Sorry about that. Seems to work fine now.

**Review by habbes — APPROVED** [review]
Looks good to me.

**Pull request merged**